oauth2_http/java/com/google/auth/oauth2/UserCredentials.java

```diff
@@ -98,6 +103,10 @@ static UserCredentials fromJson(Map<String, Object> json, HttpTransport transpor
    */
   @Override
   public AccessToken refreshAccessToken() throws IOException {
+    if (refreshToken ==  null) {
+      throw new IOException("UserCredentials instance cannot refresh because no refresh token was"
+          +" specified in constructor.");
```

Comment:
add a space after '+'. Also, you might want to avoid referring to coincidental details of the implementation, such as the fact that currently, the only path to providing the refresh token is the constructor; maybe we might add other paths to setting it later.